backend/src/activity-log/activity-log.service.ts

```diff
@@ -0,0 +1,37 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { ActivityLogDTO, PriorityENUM } from './dto/activity-log.dto';
+import { OnEvent } from '@nestjs/event-emitter';
+import { PrismaService } from '../global/prisma/prisma.service';
+import { ActivityLogGateway } from './activity-log.gateway';
+@Injectable()
+export class AnnouncementCreateListenerService {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly activityLogGateway: ActivityLogGateway
+  ) {}
+
+  @OnEvent('create.logs')
```

Comment:
used the EventEmitter api since I initially planned to use socket io and EventEmitter side by side.